### PR TITLE
Refine field edit inputs

### PR DESCRIPTION
**Pull request by johnclary**
## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/16117

## Testing
**URL to test:** [Netlify](https://deploy-preview-1390--atd-vze-staging.netlify.app/)

**Steps to test:**
1. Go to the crash page and edit a unit and person record. Notice that when you click the pencil button, the editable field is auto-focused ✅ 
2. Scroll down the page and edit some crash fields. Notice that when you hover over an editable field, the cursor is a `pointer`, you can click anywhere in the field's row to toggle editing, and field input is auto-focused ✅ 
3. Log in as a read-only user and verify that you cannot edit any fields ✅ 

---
#### Ship list
- [ ] Check migrations for any conflicts with latest migrations in master branch
- [ ] Confirm Hasura role permissions for necessary access
- [ ] Code reviewed
- [ ] Product manager approved

### CODE REVIEW

File: atd-vze/src/Components/DataTable.js [hunk 1]
```diff
@@ -149,66 +145,89 @@ const DataTable = ({
...
+                                        defaultValue={fieldValue}
+                                        type="select"
                                       >
-                                        {option[`${lookupPrefix}_desc`]}
-                                      </option>
-                                    ))}
-                                  </Input>
-                                )}
-                                {fieldUiType === "text" && (
-                                  <input
```
Comment (johnclary) [inline atd-vze/src/Components/DataTable.js:183]: Switching from a plain old `<input>` to the bootstrapped `<Input>` component. This matches the `select` component above.

**Review by roseeichelmann — APPROVED** [review]
yay really nice improvements! :sparkles: it is bothering me though that the :heavy_check_mark: :x:  boxes next to the crash fields on edit are slightly overlapping the blue autofocus line... picking nits tho :p
Screenshot from 2024-03-04 10-17-38

**Comment by johnclary** [comment]
✅ great call—thanks @roseeichelmann. how about we just go ahead and use the same button styles as in the related record table? i just made this change:
Screenshot 2024-03-04 at 2 46 11 PM

**Comment by roseeichelmann** [comment]
@johnclary muchhh better the other buttons were kind of clunky. perf :sparkles:

**Review by mddilley — APPROVED** [review]
I tested as read-only and editor+ on the deploy preview, and all works just as expected. Thanks for these really nice improvements! 🙏 ✨

**Pull request merged**